gandalf.js

```diff
@@ -1,70 +1,71 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?